lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -122,12 +122,14 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
             index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
         except MetadataError as exc:
             if Metadata.getvalue(dataset, Metadata.SERVER_ARCHIVE):
-                raise APIAbort(HTTPStatus.CONFLICT, "Dataset is not indexed")
+                raise APIAbort(HTTPStatus.CONFLICT, "Dataset indexing was disabled")
             raise APIInternalError(
                 f"Required metadata {Metadata.INDEX_MAP} missing"
             ) from exc
 
         if index_map is None:
+            if Metadata.getvalue(dataset, Metadata.SERVER_ARCHIVE):
+                raise APIAbort(HTTPStatus.CONFLICT, "Dataset indexing was disabled")
```

Comment:
This looks like an awkward repetition, assuming that the two results are supposed to be identical.  We should probably define a single object, either for the message string or for the `APIAbort` instance (or even create a subclass for it...), so that we are consistent between these two references.

And, I think I liked the "Dataset is marked for archive only" verbiage better (but, I'm OK with "disabled" if you really prefer it).

Reply:
While it would reduce the duplication, I really didn't like the idea of doing the work up front and then throwing it away when no error occurs.

Although I suppose one alternative, which perhaps I should have implemented (though it seems indirect) would be to completely ignore `index-map` if `archiveonly` is set since there *shouldn't* be an index in that case. Hmm.

But I think "indexing was disabled" is much more direct than assuming the human behind the client understands what "archive only" means.

Reply:
> While it would reduce the duplication, I really didn't like the idea of doing the work up front and then throwing it away when no error occurs.

In that case, just define a variable for the string literal and use it in both places...I doubt that does much more than the code does, now.  The obvious alternative is to define the object at module scope, so that it is created only once and never thrown away.  (Or, create a subclass of `APIAbort` and instantiate it when needed.)

> But I think "indexing was disabled" is much more direct than assuming the human behind the client understands what "archive only" means.

OK.  ("Archive only" is sort of tied to the Agent interface (or...will be, once we document all this stuff...), and, to me, it strongly implies limited functionality....)